File: Selenium/Activities/Activity12.java
package Activities;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.WebDriver;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.firefox.FirefoxDriver;


public class Activity12 {
    public static void main(String[] args) throws InterruptedException
    {
        WebDriverManager.firefoxdriver().setup();
        WebDriver newDriver = new FirefoxDriver();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE,"/dev/null");
        newDriver.get("https://training-support.net/selenium/dynamic-controls");
        System.out.println("Title is: "+newDriver.getTitle());
        WebElement textEle = newDriver.findElement(By.id("input-text"));
        System.out.println("Input enabled? " + textEle.isEnabled());
        Thread.sleep(300);
        newDriver.findElement(By.id("toggleInput")).click();
        System.out.println("Input enabled? " + textEle.isEnabled());
        Thread.sleep(300);
        newDriver.quit();

    }
}

File: Selenium/Activities/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

serv = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=serv) as driver:
    wait = WebDriverWait(driver, 10)
    driver.get("https://training-support.net/selenium/dynamic-controls")
    print("Title is: ", driver.title)
    textEle = driver.find_element(By.ID, "input-text")
    print("Is text field enabled? ", textEle.is_enabled())
    driver.find_element(By.ID,"toggleInput").click()
    print("Is text field enabled? ", textEle.is_enabled())
    driver.quit()

